checker/src/main/java/org/checkerframework/checker/calledmethods/CalledMethodsTransfer.java

```diff
@@ -0,0 +1,40 @@
+package org.checkerframework.checker.calledmethods;
+
+import org.checkerframework.common.accumulation.AccumulationTransfer;
+import org.checkerframework.dataflow.analysis.TransferInput;
+import org.checkerframework.dataflow.analysis.TransferResult;
+import org.checkerframework.dataflow.cfg.node.MethodInvocationNode;
+import org.checkerframework.dataflow.cfg.node.Node;
+import org.checkerframework.framework.flow.CFAnalysis;
+import org.checkerframework.framework.flow.CFStore;
+import org.checkerframework.framework.flow.CFValue;
+
+/** A basic transfer function that accumulates the names of methods called. */
+public class CalledMethodsTransfer extends AccumulationTransfer {
+
+    /** The type factory. */
+    private final CalledMethodsAnnotatedTypeFactory atypeFactory;
+
+    /**
+     * default constructor
+     *
+     * @param analysis the analysis
+     */
+    public CalledMethodsTransfer(final CFAnalysis analysis) {
+        super(analysis);
+        atypeFactory = (CalledMethodsAnnotatedTypeFactory) analysis.getTypeFactory();
+    }
+
+    @Override
+    public TransferResult<CFValue, CFStore> visitMethodInvocation(
+            final MethodInvocationNode node, final TransferInput<CFValue, CFStore> input) {
+        TransferResult<CFValue, CFStore> result = super.visitMethodInvocation(node, input);
+        String methodName = node.getTarget().getMethod().getSimpleName().toString();
+        methodName = atypeFactory.adjustMethodNameUsingValueChecker(methodName, node.getTree());
+        Node receiver = node.getTarget().getReceiver();
+        if (!"<init>".equals(methodName) && receiver != null) {
```

Comment:
This test against methodName seems to come too late.  For efficiency, it could occur after the assignment and before the reassignment.  More importantly, can `methodName` ever be `"<init>"`?  If the method is `<init>`, wouldn't `node` have to be a ConstructorInvocationNode?

Also, the reassignment could occur within the `receiver != null` test.

Reply:
> If the method is <init>, wouldn't node have to be a ConstructorInvocationNode

That's what I would expect. I thought this had caused a failure when I was testing the accumulation checker, but I can't reproduce it. I'll remove the test.

> Also, the reassignment could occur within the receiver != null test.

I think that is bad style.